sn_transfers/src/wallet/watch_only.rs

```diff
@@ -106,11 +106,6 @@ impl WatchOnlyWallet {
         for cash_note in received_cash_notes {
             let id = cash_note.unique_pubkey();
 
-            if self.keyless_wallet.spent_cash_notes.contains(&id) {
```

Comment:
this maybe a strong reason to keep `spent_cash_notes` to avoid deposit cash_notes twice ?
or there is other way to prevent so ?

Reply:
I'd vote for throw an error in deposit code (when such duplicated deposit is detected by some means)

Reply:
I'd call that out of scope for this PR, I think? 

This check would only work for persistent wallets as is, so that's a feat we'll need properly (if it's not already there..) ill check and make an issue if no

Reply:
https://github.com/maidsafe/safe_network/issues/1091